unlock-app/src/components/interface/locks/Settings/forms/UpdateReferralFee.tsx

```diff
@@ -16,82 +18,115 @@ interface UpdateReferralFeeProps {
 
 interface FormProps {
   referralFeePercentage: number
+  referralAddress: string
 }
 
-const ZERO = ethers.constants.AddressZero
+interface ReferrerFee {
+  id: string
+  referrer: string
+  fee: string
+}
 
 export const UpdateReferralFee = ({
   lockAddress,
   network,
   isManager,
   disabled,
 }: UpdateReferralFeeProps) => {
-  const [referralFee, setReferralFee] = useState(false)
-  const web3Service = useWeb3Service()
+  const [isReferralFeeEnabled, setIsReferralFeeEnabled] = useState(false)
+  const [isReferralAddressEnabled, setIsReferralAddressEnabled] =
+    useState(false)
+  const [isResolvingAddress, setIsResolvingAddress] = useState(false)
   const { getWalletService } = useAuth()
 
   const {
     register,
     handleSubmit,
-    setValue,
+    reset,
     formState: { errors },
-  } = useForm<FormProps>()
+  } = useForm<FormProps>({
+    defaultValues: { referralFeePercentage: 0, referralAddress: '' },
+  })
 
   const setReferrerFee = async (fields: FormProps) => {
     const walletService = await getWalletService(network)
     await walletService.setReferrerFee({
       lockAddress,
-      address: ZERO,
+      address: fields?.referralAddress,
       feeBasisPoint: fields?.referralFeePercentage * 100,
     })
   }
 
-  const getReferrerFees = async () => {
-    return await web3Service.referrerFees({
-      lockAddress,
-      network,
-      address: ZERO,
-    })
+  const getLock = async () => {
```

Comment:
Here we will get the `lock` and then the lock object will have the `referrer` array